heudiconv/cli/run.py

```diff
@@ -96,14 +96,19 @@ def process_extra_commands(outdir, args):
     return
 
 
-def main(argv=None):
+def main(args=None):
     parser = get_parser()
-    args = parser.parse_args(argv)
-    # exit if nothing to be done
-    if not args.files and not args.dicom_dir_template and not args.command:
-        lgr.warning("Nothing to be done - displaying usage help")
-        parser.print_help()
-        sys.exit(1)
+    argv = sys.argv[1:]
```

Comment:
not yet clear to me why we need to overload those and why it all should make a difference. args should be the args to be parsed by the parser, either they come from the tests or cmdline (via `sys.argv`). `main` shouldn't get the args as a structure to be overwritten later on etc

Reply:
Ok I may be entirely wrong and the those changes are not required. Its all abit confusing when debugging the regression test test_conversion (I've modified this slightly, posted below). Perhaps it could be changed simply to avoid that. Here's what I observe:

+ Args exists as an object from pytest at the start of the test. Printing this object fails.
+ After  "args = gen_heudiconv_args(datadir, outdir, subject, heuristic)" is executed print now works and outputs a list.
+ Stepping into runner(args), args is now:

```
 argv = ['-d', '/tmp/pytest-of-root/QA/sourcedata/{subject}/*/*/*.tgz', '-c', 'dcm2niix', '-o', '/tmp/pytest-of-root/out', '-s', 'sub-sid000143', '-f', '/src/heudiconv/heudiconv/heuristics/reproin.py', '--bids']
 ```

Running parser.parse_args(argv) outputs the namespace object we want but running 

```
args = parser.parse_args(argv)
```
 Does not assign the result to args. Subsequent tests using its attributes are not really going to work for example  bool( args.grouping) at this time will yield true.

Do you observe this?



### test_conversion:
```
"""Testing conversion with conversion saved on datalad"""
import json
from glob import glob

import pytest

have_datalad = True
try:
    from datalad import api # to pull and grab data
    from datalad.support.exceptions import IncompleteResultsError
except ImportError:
    have_datalad = False

import heudiconv
from heudiconv.cli.run import main as runner
# testing utilities
from .utils import fetch_data, gen_heudiconv_args
import py
import os
import shutil

@pytest.mark.parametrize('tmpdir', [py._path.local.LocalPath('/tmp/pytest-of-root')])
@pytest.mark.parametrize('datadir', ['pytest-3'])
@pytest.mark.parametrize('subject', ['sub-sid000143'])
@pytest.mark.parametrize('heuristic', ['reproin.py'])
@pytest.mark.skipif(not have_datalad, reason="no datalad")
def test_conversion(tmpdir, subject, heuristic,datadir):
    tmpdir.chdir()
    if os.path.exists('out'):
        shutil.rmtree('out')
    if os.path.exists(datadir):
        outdir = os.path.join(tmpdir.strpath, 'out')
    else:
        try:
            datadir = fetch_data(tmpdir.strpath, subject)
        except IncompleteResultsError as exc:
            pytest.skip("Failed to fetch test data: %s" % str(exc))
        outdir = tmpdir.mkdir('out').strpath

    import pdb; pdb.set_trace()
    args = gen_heudiconv_args(datadir, outdir, subject, heuristic)
    print(args)
    # import pdb;pdb.set_trace()
    runner(args) # run conversion

    # verify functionals were converted
    assert glob('{}/{}/func/*'.format(outdir, subject)) == \
           glob('{}/{}/func/*'.format(datadir, subject))

    # compare some json metadata
    json_ = '{}/task-rest_acq-24mm64sl1000tr32te600dyn_bold.json'.format
    orig, conv = (json.load(open(json_(datadir))),
                  json.load(open(json_(outdir))))
    keys = ['EchoTime', 'MagneticFieldStrength', 'Manufacturer', 'SliceTiming']
    for key in keys:
        assert orig[key] == conv[key]
```

Reply:
Ok part of the confusion is that args is a pdb command. that may solve it

Reply:
naming it something other than other args would be helpful for any future pdb debugging sessions. I forgot args was a pdb command. But yes provided you aren't stepping through the code with pdb it all makes sense.

Reply:
I didn't even know about args pdb command ;-)